OpenRA.Game/Actor.cs

```diff
@@ -454,6 +511,70 @@ public IEnumerable<WPos> GetTargetablePositions()
 			return new[] { CenterPosition };
 		}
 
+		#region Conditions
+
+		void UpdateConditionState(string condition, int token, bool isRevoke)
+		{
+			ConditionState conditionState;
+			if (!conditionStates.TryGetValue(condition, out conditionState))
+				return;
+
+			if (isRevoke)
+				conditionState.Tokens.Remove(token);
+			else
+				conditionState.Tokens.Add(token);
+
+			conditionCache[condition] = conditionState.Tokens.Count;
+
+			// Wait until all traits are created.
+			if (created)
+				foreach (var notify in conditionState.Notifiers)
+					notify(this, readOnlyConditionCache);
+		}
+
+		/// <summary>Grants a specified condition.</summary>
+		/// <returns>The token that is used to revoke this condition.</returns>
+		public int GrantCondition(string condition)
+		{
+			var token = nextConditionToken++;
+			conditionTokens.Add(token, condition);
+
+			// Conditions may be granted before the state is initialized.
+			// These conditions will be processed in INotifyCreated.Created.
+			if (conditionStates != null)
```

Comment:
How can this be `null`? (I.e. this will never be false?)

Reply:
It's left-overs from copy-pasting.